pkg/server/exec/handlers.go

```diff
@@ -0,0 +1,49 @@
+package exec
+
+import (
+	"github.com/luizalabs/teresa/pkg/goutil"
+	execpb "github.com/luizalabs/teresa/pkg/protobuf/exec"
+	"github.com/luizalabs/teresa/pkg/server/database"
+	"google.golang.org/grpc"
+)
+
+type Service struct {
+	ops Operations
+}
+
+func (s *Service) Command(req *execpb.CommandRequest, stream execpb.Exec_CommandServer) error {
+	ctx := stream.Context()
+	u := ctx.Value("user").(*database.User)
+
+	rc, errChan := s.ops.Command(u, req.Name, req.Command...)
+	if rc == nil {
+		return <-errChan
+	}
+	defer rc.Close()
+
+	cmdMsgs := goutil.ChannelFromReader(rc, true)
+	var msg string
+	for {
+		select {
+		case err := <-errChan:
+			return err
+		case m, ok := <-cmdMsgs:
+			if !ok {
+				return nil
+			}
+			msg = m
+		}
+
+		if err := stream.Send(&execpb.CommandResponse{Text: msg}); err != nil {
+			return err
+		}
+	}
+}
```

Comment:
Maybe add `ctx.Done()` but I don't know the complexity so follow your heart.

Reply:
Good point, I'll move the logic of `ctx.Done()` + `cancelation` of pkg deploy to pkg exec, what do you think about it?

Reply:
Excelent.